primitives/src/identity_claim.rs

```diff
@@ -158,7 +163,7 @@ impl Claim {
 /// Claim type represent the claim without its data.
 ///
 /// # TODO
-/// - Could we use `std::mem::Discriminat`?
+/// - Could we use `std::mem::Discriminant`?
```

Comment:
`mem::Discriminant` is not something you can match on, but `PartialEq` of two `Claim`s is sufficient, then yes, you can.